src/test/credentials/diskCache.test.ts

```diff
@@ -0,0 +1,187 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as path from 'path'
+import { openSync } from 'fs'
+import { fstatSync, writeFileSync } from 'fs-extra'
+import { fileExists, makeTemporaryToolkitFolder } from '../../shared/filesystemUtilities'
+import { DiskCache } from '../../credentials/sso/diskCache'
+import { SsoClientRegistration } from '../../credentials/sso/ssoClientRegistration'
+import { rmrf } from '../../shared/filesystem'
+import { SsoAccessToken } from '../../credentials/sso/ssoAccessToken'
+
+describe('SSO diskCache', () => {
+    let tempFolder: string
+    const ssoRegion = 'dummyRegion'
+    const startUrl = 'https://123456.awsapps.com/start'
+    let registrationFilename: string
+    let accessTokenFileName: string
+    const HOUR_IN_MS = 3600000
+    const FOURTEEN_MINS_IN_MS = 840000
+    let sut: DiskCache
+
+    const validRegistration: SsoClientRegistration = {
+        clientId: 'dummyId',
+        clientSecret: 'dummySecret',
+        expiresAt: new Date(Date.now() + HOUR_IN_MS).toISOString(),
+    }
+
+    const validAccessToken: SsoAccessToken = {
+        startUrl: startUrl,
+        region: ssoRegion,
+        accessToken: 'longstringofrandomcharacters',
+        expiresAt: new Date(Date.now() + HOUR_IN_MS).toISOString(),
+    }
+
+    beforeEach(async () => {
+        // Make a temp folder for all these tests
+        tempFolder = await makeTemporaryToolkitFolder()
+        registrationFilename = path.join(tempFolder, `aws-toolkit-vscode-client-id-${ssoRegion}.json`)
+        accessTokenFileName = path.join(tempFolder, 'c1ac99f782ad92755c6de8647b510ec247330ad1.json')
```

Comment:
Nit: This is oddly specific, any reason why?

Reply:
Yes, the file name is the string hash of the encoded start url.  So that is directly linked to the startUrl const above.